### ISSUE TITLE
Error when flushing

### ISSUE THREAD

**Issue by n-batalha** [transferwise/pipelinewise-target-snowflake#41]
## Log

```
INFO Deleting <REDACTED>.csv from external snowflake stage on S3
Traceback (most recent call last):
  File "/Users/nqb/.local/share/virtualenvs/job-KgtlUjJP/bin/target-snowflake", line 11, in <module>
    load_entry_point('pipelinewise-target-snowflake', 'console_scripts', 'target-snowflake')()
  File "/Users/nqb/Documents/git/extract-<REDACTED>/pipelinewise-target-snowflake/target_snowflake/__init__.py", line 393, in main
    persist_lines(config, singer_messages, information_schema_cache)
  File "/Users/nqb/Documents/git/extract-<REDACTED>/pipelinewise-target-snowflake/target_snowflake/__init__.py", line 182, in persist_lines
    filter_streams=filter_streams)
  File "/Users/nqb/Documents/git/extract-<REDACTED>/pipelinewise-target-snowflake/target_snowflake/__init__.py", line 328, in flush_streams
    if stream in state.get('bookmarks', {}):
AttributeError: 'NoneType' object has no attribute 'get'
```

## Environment

Using a custom built tap with `master` of this repo (I've moved back to v.1.0.9 where this doesn't happen) and Python 3.7.

## More info

Skimming the code, it doesn't seem to depend on the environment, it seems to happen everytime?

`state` is `None` in https://github.com/transferwise/pipelinewise-target-snowflake/blob/master/target_snowflake/__init__.py#L103, then it is unchanged until `state.get` is called in https://github.com/transferwise/pipelinewise-target-snowflake/blob/56c302849180e565716de30716757435ffa8b6c1/target_snowflake/__init__.py#L328

so `AttributeError: 'NoneType' object has no attribute 'get'`?

**Comment by koszti** [comment]
This is a bug and #42 fixing it. Once this PR got merged we will create a new package of this connector on PyPI.

**Some more background:**
Since 1.1.0 the state messages are emitted more often, every time when a batched data flushed and the connector expects that at least one STATE message received from the tap by this time. This is true for the currently supported pipelinewise taps but it's not mandatory by the singer spec and should not be required for any other custom built taps.

**Quick fix until the new package is not on PyPI:**
Modify your custom tap to send a valid singer STATE message to the target before start extracting data.

**Comment by koszti** [comment]
fixed by #42. v1.1.4 released to PyPI (https://pypi.org/project/pipelinewise-target-snowflake/1.1.4/)

**Comment by koszti** [comment]
1.1.4 was having some dependency issues with the latest snowflake-connector-python. Fixed an released 1.1.5 to PyPI (https://pypi.org/project/pipelinewise-target-snowflake/1.1.5/)

**Comment by n-batalha** [comment]
thanks!
